### PR TITLE
Testing script for image display in main window

### LINKED ISSUE #31: Testing Scripts creation
Write and integrate a test suite for the GUI using an appropriate testing framework. These tests should cover:

 

- [x] GUI initialization and rendering

- [x]  Button clicks and their callbacks

- [x]  Input field behaviors and validations

- [ ]  Navigation between GUI panels

- [x]  Output/logging from user interaction
- [ ] Display the telemetry information
- [x] Dispaly the graphs and images from the api

- [x]  Edge cases (e.g., invalid inputs, missing fields)

Tests should be runnable using pytest or a GUI testing tool like pytest-qt, unittest, or PyAutoGUI, depending on the framework used.

Include clear documentation on how to run the tests.

Tests should be structured for maintainability and reusability.

there will be someonelese who will be working on creating the pipelines and the dockerization

### PR DESCRIPTION
**Pull request by samdeupta**
Automated testing script for image display component in `main_window.py`.

### CODE REVIEW

**Comment by samdeupta** [comment]
**Manual Testing Results:**

- The image display successfully showed the latest image added to the monitored folder.
- When a corrupted image was added, the display correctly showed the fallback message (`"Error loading image"`).

**Comment by samdeupta** [comment]
Additionally, when a valid `.jpg` was added after a corrupt file, the display recovered and showed the correct image as expected.

**Review by AniMB — APPROVED** [review]
looks good for testing. will recommend @yeshwanthkonk to have a look as well

**Review by yeshwanthkonk — CHANGES_REQUESTED** [review]
Test pipelines are now available,
For now, update the PR by resolving the conflicts, which will trigger the pipeline to validate the test cases. PR should pass the pipeline to be merged.

**Comment by AniMB** [comment]
There is one bug that has to be rectified. Will push once bug is rectified

**Comment by ABhiShek1705m** [comment]
> There is one bug that has to be rectified. Will push once bug is rectified

@AniMB which bug are we talking about here?

**Comment by AniMB** [comment]
@ABhiShek1705m The utils.py file for the CommandPrompt dropdown menu hasn't implemented the global transfer_acknowledgement flag yet. So tests which check the flag are failing.

**Comment by ABhiShek1705m** [comment]
@samdeupta can you resolve the conflicts and merge in if you're done making changes.

**Review by yeshwanthkonk — COMMENTED** [review]
Take pull from dev and update this PR after the pipeline fix PR(https://github.com/UVicSatelliteDesign/COMP_SLIP-GS/pull/67) is merged, make sure this PR passes the pipeline.

**Comment by samdeupta** [comment]
The pipeline is failing because one of the test cases is not passing. Nothing to do with the testing scripts. `utils.py` has not implemented the global `command_queue_state` flag properly.

**Comment by samdeupta** [comment]
`main.py` is using the old implementation of `command_queue`.

File: front_end_tests/unit_tests/test_main_window_image_display.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+from PyQt6.QtGui import QImage
+
+# Widget to test
+from ground_station.frontend.main_window import GroundStationMainWindow
+
+DIR_PATH = Path("front_end_tests") / "unit_tests" / "imgs"  # Path for directory in which test images are created
+DEFAULT_MAX_IMGS = 50   # Default upper-limit for no. of images used in testing
+
+# Hook for optional user input of custom upper-limit for the no. of images to test with
+def pytest_addoption(parser: pytest.Parser):
```
Comment (yeshwanthkonk) [inline front_end_tests/unit_tests/test_main_window_image_display.py]: `parser.addoption` is already defined in the conftest file, so it's redundant here and can be removed. If not being updated in this PR, please make a note to remove it in the next one.

File: front_end_tests/conftest.py [hunk 2]
```diff
@@ -6,4 +6,8 @@ def pytest_addoption(parser):
     parser.addoption(
```
Comment (yeshwanthkonk) [inline front_end_tests/conftest.py:6]: To maintain consistency, we could convert all argument names from hyphens (`-`) to underscores (`_`) for `--max-commands` and `--max-graphs`. If not updating it now, please make a note to address it in the next PR.

**Review by yeshwanthkonk — APPROVED** [review]
I’ve left a couple of comments, not blockers, but they can be addressed later as well. Please take a look when you get a chance.

**Review by yeshwanthkonk — CHANGES_REQUESTED** [review]
The pipeline failed on this commit and needs fixing before merging.

**Comment by samdeupta** [comment]
Reverted pipeline breaking changes. To be fixed later.

**Review by yeshwanthkonk — APPROVED** [review]
Nice, keep a note of all required changes.
